lib/seahorn/BvOpSem2Context.hh

```diff
@@ -553,7 +554,7 @@ public:
   virtual PtrSortTy mkPtrRegisterSort(const GlobalVariable &gv) const = 0;
 
   /// \brief Returns sort of memory-holding register for an instruction
-  virtual MemSortTy mkMemRegisterSort(const Instruction &inst) const = 0;
+  virtual Expr mkMemRegisterSort(const Value &value) const = 0;
```

Comment:
return type should be `MemSortTy`. Comment needs to be updated because it talks about instruction.
I would prefer to to not have `mkMemRegisterSort` for `Value`. It is easy to abuse and get wrong

Reply:
Ok, I see why you are using `Value`. Add assertions to make sure that only `MemSSA` or `Instruciton` are allowed as arguments and nothing else.